test/donotoptimize_assembly_test.cc

```diff
@@ -38,88 +38,100 @@ const ExtraLarge ConstExtraLargeObj{};
 // CHECK-LABEL: test_with_rvalue:
 extern "C" void test_with_rvalue() {
   benchmark::DoNotOptimize(Add42(0));
-  // CHECK: movl $42, %eax
-  // CHECK: ret
+  // CHECK-CLANG: mov 2a,REG
+  // CHECK-CLANG: ret
 }
 
 // CHECK-LABEL: test_with_large_rvalue:
 extern "C" void test_with_large_rvalue() {
   benchmark::DoNotOptimize(Large{ExternInt, {ExternInt, ExternInt}});
-  // CHECK: ExternInt(%rip)
-  // CHECK: movl %eax, -{{[0-9]+}}(%[[REG:[a-z]+]]
-  // CHECK: movl %eax, -{{[0-9]+}}(%[[REG]])
-  // CHECK: movl %eax, -{{[0-9]+}}(%[[REG]])
+  // CHECK: mov OFFSET(RIP),REG
+  // CHECK-NVHPC: mov (REG),REG
+  // CHECK: mov REG,OFFSET(REG)
+  // CHECK: mov REG,OFFSET(REG)
+  // CHECK: mov REG,OFFSET(REG)
   // CHECK: ret
 }
 
 // CHECK-LABEL: test_with_non_trivial_rvalue:
 extern "C" void test_with_non_trivial_rvalue() {
   benchmark::DoNotOptimize(NotTriviallyCopyable(ExternInt));
-  // CHECK: mov{{l|q}} ExternInt(%rip)
+  // CHECK: mov OFFSET(RIP),REG
```

Comment:
Here the normalization comes handy as we do not have to use a regex to match the mov variants

Reply:
I reverted this